superset/jinja_context.py

```diff
@@ -722,3 +723,69 @@ def dataset_macro(
     sqla_query = dataset.get_query_str_extended(query_obj, mutate=False)
     sql = sqla_query.sql
     return f"(\n{sql}\n) AS dataset_{dataset_id}"
+
+
+def get_dataset_id_from_context(metric_key: str) -> int:
+    """
+    Retrives the Dataset ID from the request context.
+
+    :param metric_key: the metric key.
+    :returns: the dataset ID.
+    """
+    # pylint: disable=import-outside-toplevel
+    from superset.daos.chart import ChartDAO
+    from superset.views.utils import get_form_data
+
+    exc_message = _(
+        "Please specify the Dataset ID for the ``%(name)s`` metric in the Jinja macro.",
+        name=metric_key,
+    )
+    if form_data := get_form_data()[0]:
```

Comment:
nit: you could:
``` python
if not form_data, _ := get_form_data():
    raise SupersetTemplateException(exc_message)
...
```
to reduce nesting

btw, would the `slice_id` optionally returned by `get_form_data` help here?

Reply:
Thanks for the suggestion! Just added a commit taking advantage of the `Slice` returned from `get_form_data()[1]` and also reducing nesting.